plugins/woocommerce-blocks/assets/js/atomic/blocks/product-elements/related-products/index.tsx

```diff
@@ -11,14 +11,14 @@ import edit from './edit';
 import save from './save';
 import metadata from './block.json';
 
-registerProductBlockType( {
-	blockName: metadata.name,
-	// @ts-expect-error: `metadata` currently does not have a type definition in WordPress core
-	blockMetadata: metadata,
-	blockSettings: {
-		icon,
-		edit,
-		save,
-	},
+const blockConfig = {
+	...metadata,
+	icon: { src: icon },
+	edit,
+	save,
+	isAvailableOnPostEditor: false,
+};
+
+registerProductBlockType( blockConfig, {
 	isAvailableOnPostEditor: false,
 } );
```

Comment:
`isAvailableOnPostEditor` is `false` by default. I think we can simplify these declarations even more and remove the settings wherever `isAvailableOnPostEditor: false` is passed.

```suggestion
registerProductBlockType( blockConfig );
```

Reply:
We _could_ do but I intentionally left that in so it's more declarative, but good spot!

Reply:
PS. Im inclined to leave it like it is now, so its limitations are clear.

Reply:
Yeah I saw you even add it where is wasn't there initially. I don't have a strong opinion. Actually I get it and agree. But then, maybe it should not be optional. Because it seems it's not 😄 It's a non-blocking detail though.

Reply:
Yeah good point about it being optional. This setting was historical optional so I kept it that way to simply the refactor for now.